src/main/java/com/thebois/models/inventory/AbstractInventory.java

```diff
@@ -0,0 +1,85 @@
+package com.thebois.models.inventory;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+import com.thebois.models.inventory.items.IItem;
+import com.thebois.models.inventory.items.ItemType;
+
+/**
+ * Base implementation for all inventories.
+ */
+public abstract class AbstractInventory {
```

Comment:
I think it would be a great idea to have an interface for Inventories, so that we don't have to depend on any specific Type  of class

Reply:
Thats true, thought since its only one class it was OK, but either way its better.